### ISSUE TITLE
The check for updates command line/configuration setting on the `lint` and `analyze` commands, no update check is performed

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#5904]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

When using the `--check-for-updates` command line option or configuration file setting on the `lint` and `analyze` commands, no update is performed, because we `exit(2)` once we've determined that violations are present

Create a file with some violations. For example

```swift
let foo = try! bar()
```
Then, run something like:

```bash
% swiftlint.debug lint --check-for-updates  t.swift
Linting Swift files at paths t.swift
Linting 't.swift' (1/1)
/tmp/t.swift:2:11: error: Force Try Violation: Force tries should be avoided (force_try)
Done linting! Found 1 violation, 1 serious in 1 file.
%
% echo $?
2
```

We should see a message relating to the update check as well, such as

```
Your version of SwiftLint is up to date.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)

`0.57.1`

* Xcode version (run `xcodebuild -version` to be sure)

16.1

* Installation method used (Homebrew, CocoaPods, building from source, etc)

Cocoa Pods
